src/test_l2_dsCompositeIn.c

```diff
@@ -1,119 +1,307 @@
+    UT_LOG("Invoked dsCompositeInGetStatus(). Return status: %d ,port status.isPresented : %d, isPortConnected[%d] : %d, activePort: %d ", ret, status.isPresented, port, status.isPortConnected[port], status.activePort);
+    UT_ASSERT_EQUAL(ret, dsERR_NONE);
+    if (ret != dsERR_NONE)
+    {
+        UT_LOG("dsCompositeInGetStatus failed with error: %d", ret);
+        dsCompositeInTerm();
+        return;
+    }
+
+    // Check the returned status
+    UT_ASSERT_EQUAL(status.isPresented, false);
+    UT_ASSERT_EQUAL(status.isPortConnected[port], false);
+    UT_ASSERT_EQUAL(status.activePort, port);
+
+    // Terminate the COMPOSITE Input sub-system
+    ret = dsCompositeInTerm();
+    UT_ASSERT_EQUAL(ret, dsERR_NONE);
+    if (ret != dsERR_NONE)
+    {
+        UT_LOG("dsCompositeInTerm failed with error: %d", ret);
+    }
+
+    UT_LOG("Exiting test_l2_dsCompositeIn_SetAndCheckCompositeInputPort...");
+}
 
 /**
-* @brief TODO: Describe the object of the test
+* @brief Test for scaling video without selecting COMPOSITE Input port
 *
-* TODO: Add the description of what is tested and why in this test
+* This function tests the scenario where `dsCompositeInScaleVideo` is called without selecting the COMPOSITE Input port. It first initializes the COMPOSITE Input module using `dsCompositeInInit`, then tries to scale the video using `dsCompositeInScaleVideo`, and finally terminates the COMPOSITE Input module using `dsCompositeInTerm`.
 *
-* **Test Group ID:** TODO: Add the group this test belongs to - Basic (for L1): 01 / Module (L2): 02 / Stress (L2): 03)@n
-* **Test Case ID:** TODO: Add the ID of the test case so that it can be logically tracked in the logs@n
+* **Test Group ID:** 02@n
+* **Test Case ID:** 004@n
 *
 * **Test Procedure:**
-* TODO: Refer to L2 Specification documentation : [l2_module_test_specification_template.md](../docs/pages/l2_module_test_specification_template.md)
+* Refer to UT specification documentation [l2_dsCompositeIn_test_specification.md](l2_dsCompositeIn_test_specification.md)
 */
-void test_l2_dsCompositeIn (void)
+
+void test_l2_dsCompositeIn_ScaleVideoWithoutSelectingPort(void)
 {
-	UT_FAIL("This function needs to be implemented!");
+    UT_LOG("Entering test_l2_dsCompositeIn_ScaleVideoWithoutSelectingPort...");
+
+    dsError_t status;
+    int32_t x, y, width, height;
+
+    srand(time(0)); // use current time as seed for random generator
+    x = rand() % 1000; // assuming the resolution width is 1000
+    y = rand() % 1000; // assuming the resolution height is 1000
+    width = rand() % (1000 - x); // width should not exceed the resolution width
+    height = rand() % (1000 - y); // height should not exceed the resolution height
+
+    status = dsCompositeInInit();
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+    if (status != dsERR_NONE)
+    {
+        UT_LOG("dsCompositeInInit failed with status: %d", status);
+        return;
+    }
+
+    status = dsCompositeInScaleVideo(x, y, width, height);
```

Comment:
dsCompositeInSelectPort is pre requisite  for dsCompositeInScaleVideo. But this API not called

Reply:
This testcase is expected to call dsCompositeInScaleVideo without calling dsCompositeInSelectPort api.